types/use-location.d.ts

```diff
@@ -26,16 +26,25 @@ export type HookNavigationOptions<H extends BaseLocationHook> = HookReturnValue<
     : {}
   : {};
 
+type Primitive = string | number | bigint | boolean | null | undefined | symbol;
+export const useLocationProperty: <S extends Primitive>(fn: () => S) => S;
+
+export const useSearch: () => string;
+
+export const usePathname: () => Path;
+
+export type Navigate = (to: Path, options?: { replace?: boolean }) => void;
+
+export const useNavigate: (options?: { base?: Path }) => Navigate;
```

Comment:
@molefrog only thing I'm not sure on... am I supposed to use `export const` or `declare const` here (like was done below for `useLocation`)? I don't know what the difference is. (And I can't even find any documentation on what `declare const` does 😅 ).

Reply:
Good question. Based on what I was able to find, there is no actual difference between these two in a declaration file. There would be though if it was written in a source file. I think we stick to `export const` which is more correct in my opinion.

Reply:
@molefrog cool, in that case, I think this is ready to be merged in if you don't have any further feedback! I'm happy with it.